admin-ui/plugins/scim/components/ScimConfiguration.tsx

```diff
@@ -7,53 +7,70 @@ import GluuCommitDialog from 'Routes/Apps/Gluu/GluuCommitDialog'
 import GluuCommitFooter from 'Routes/Apps/Gluu/GluuCommitFooter'
 import GluuSelectRow from 'Routes/Apps/Gluu/GluuSelectRow'
 import GluuToogleRow from 'Routes/Apps/Gluu/GluuToogleRow'
-import PropTypes from 'prop-types'
 import { adminUiFeatures } from 'Plugins/admin/helper/utils'
+import type {
+  ScimConfigurationProps,
+  RootStateWithScim,
+  ScimFormValues,
+  ScimConfigPatchRequest,
+} from './types'
 
-const ScimConfiguration = ({ handleSubmit }) => {
-  const scimConfigs = useSelector((state) => state.scimReducer.scim)
-  const [modal, setModal] = useState(false)
-  const toggle = () => {
+const ScimConfiguration: React.FC<ScimConfigurationProps> = ({ handleSubmit }) => {
+  const scimConfigs = useSelector((state: RootStateWithScim) => state.scimReducer.scim)
+  const [modal, setModal] = useState<boolean>(false)
+
+  const toggle = (): void => {
     setModal(!modal)
   }
 
-  const formik = useFormik({
-    initialValues: scimConfigs,
+  const formik: FormikProps<ScimFormValues> = useFormik<ScimFormValues>({
+    initialValues: scimConfigs as ScimFormValues,
     onSubmit: () => {
       toggle()
     },
   })
 
-  const submitForm = (userMessage) => {
-    const differences = []
+  const submitForm = (userMessage: string): void => {
+    const differences: ScimConfigPatchRequest[] = []
     delete formik.values?.action_message
```

Comment:
Avoid mutatingformik.values with 'delete'. This mutation can cause unintended side effects. Consider filtering out 'action_message' in a non-mutative way (e.g. using object spread).